hs_core/views/resource_folder_hierarchy.py

```diff
@@ -167,6 +168,15 @@ def data_store_folder_zip(request):
     )
 
 
+@api_view(['POST'])
+def data_store_folder_zip_public(request, pk):
+    return data_store_folder_zip(request,
+                                 res_id=pk,
+                                 input_coll_path=request.data.get('input_coll_path', None),
+                                 output_zip_file_name=request.data.get('output_zip_file_name', None),
+                                 remove_original_after_zip=request.data.get('remove_original_after_zip', None))

```

Comment:
@aphelionz Same question as above, I don't quite understand why additional parameters need to be included if they are already included in request object?